### ISSUE TITLE
contrary to doc header, enum.Flag is inadequate

### ISSUE THREAD

**Issue by belm0** [pasztorpisti/py-flags#8]
> Python 3.6+ users should consider using the Flag and IntFlag classes of the standard enum module. Those are very similar to the flags.Flags class of this repo.

I don't agree that they are very similar.  I found py-flags after frustration with enum.Flag.  The main problem with enum.Flag is that it doesn't act sufficiently like a set.

  * 0 flag, compound flags, and aliases are included in iteration
  * **class instances lack iteration, len, and subset operator**

**Comment by pasztorpisti** [comment]
I don't agree with some of the design decisions of the standard flags but that it still the standard (and builtin) so people should consider using it when it covers their use cases. At the same time that module has a python2 port (called `aenum`) while this is python3 only.

In my opinion the standard module lacks some useful features while my solution is too feature rich. In the future I might compile a list of differences between the two.

Certain aspects of a flags library don't fit into the python language. The language itself doesn't support enums/flags (being a small language isn't necessarily a bad thing...) and some flags related features can't be simulated nicely (e.g.: you have to write `enum_member = ()` to auto-assign a value). For this reason, and because most programs don't really need bit flags (it's a relatively narrow-purpose thing) I'm wondering whether it's a good idea to have a flags as part of the standard library.

**Comment by belm0** [comment]
I was surprised that there doesn't seem to be a PEP-like process for stdlib additions.  You just open an issue with some patches, and a few people on a thread have some debate, and then... Flag and IntFlag?

One subtle thing I discovered about Flag vs. working with Enum sets:  Flag instances have the mysterious ability to represent either a single item or a collection of items.  For example, you can use the `&` operator in both cases.  With `set`, you can only use `&` with another set, and single item must use the `in` operator.

**Comment by pasztorpisti** [comment]
`Flag` vs `IntFlag`: overcomplicated - it's enough to have an integer flag with the ability to auto-assign flags.

> One subtle thing I discovered about Flag vs. working with Enum sets: Flag instances have the mysterious ability to represent either a single item or a collection of items.

All declared flag members (declared and synthesised) and all of their combinations are instances of the declared flag type (and the flag type behaves like a set). This is the most practical way I know to implement type safe flags.

Enums and flags are somewhat related but the correct relation isn't inheritance. I really like how the pascal language allows defining set/flag types from existing enum declarations and then it allows testing against the set instances using the enum members:
```pascal
TSpeed = (spVerySlow,spSlow,spAVerage,spFast,spVeryFast);
TPossibleSpeeds = set of TSpeed
```
More on this: https://wiki.freepascal.org/Set

I was thinking about mimicking pascal sets in python but in most cases it would simply make it more difficult for people to understand how the library works. Implementing such a thing as a library versus having it integrated into the language like in case of pascal are two different things.

**Comment by flying-sheep** [comment]
> - 0 flag, compound flags, and aliases are included in iteration
> - class instances lack iteration, len, and subset operator

None of this is true anymore since Python 3.11:

```pycon
In [1]: import enum
In [2]: class TextStyle(enum.Flag):
   ...:     bold = enum.auto()
   ...:     italic = enum.auto()
   ...:     underline = enum.auto()
   ...:

In [3]: list(TextStyle)
Out[3]: [<TextStyle.bold: 1>, <TextStyle.italic: 2>, <TextStyle.underline: 4>]

In [4]: list(TextStyle.italic | TextStyle.bold)
Out[4]: [<TextStyle.bold: 1>, <TextStyle.italic: 2>]

In [5]: len(TextStyle.bold | TextStyle.italic)
Out[5]: 2

In [6]: (TextStyle.bold | TextStyle.italic) in ~TextStyle(0)
Out[6]: True
```

The only problem I see is that `IntEnum` happily allows someone accustomed to sets to use `-` but returns an integer instead of a set difference.

**Comment by belm0** [comment]
> None of this is true (anymore):

yes, because I pushed the enum maintainer to make these changes

**Comment by flying-sheep** [comment]
Awesome, thanks for that!

Good thing it was possible to add everything while preserving backwards compatibility.
